### PR TITLE
feat(proxy-host-header): add proxy host header, remove PSL

### PR DESCRIPTION
**Pull request by sshelomentsev**

### CODE REVIEW

File: proxy/app.ts [hunk 1]
```diff
@@ -57,7 +56,7 @@ export const handler = async (event: CloudFrontRequestEvent): Promise<CloudFront
     if (suffix.length > 0 && !suffix.startsWith('/')) {
       suffix = '/' + suffix
     }
-    const eTLDPlusOneDomain = getEffectiveTLDPlusOne(getHost(request))
+    const eTLDPlusOneDomain = getHost(request)
```
Comment (necipallef) [inline proxy/app.ts:59]: Do we still need this variable `eTLDPlusOneDomain`?

File: proxy/test/utils/headers.test.ts [hunk 2]
```diff
@@ -70,6 +228,7 @@ describe('test fpjs-headers preparation', () => {
     const headers = await prepareHeadersForIngressAPI(req, getCustomerVariables(req))
     expect(headers['fpjs-proxy-client-ip']).toBe('1.1.1.1')
     expect(headers.hasOwnProperty('fpjs-proxy-secret')).toBeFalsy()
+    expect(headers['fpjs-proxy-forwarded-host']).toBe('fpjs.sh')
```
Comment (necipallef) [inline proxy/test/utils/headers.test.ts:231]: Maybe use more generic value, such as `foo.bar` instead of `fpjs.sh`?

**Review by necipallef — APPROVED** [review]

**Review by ilfa — APPROVED** [review]

**Pull request merged**